### ISSUE TITLE
Error when cancelling build

### ISSUE THREAD

**Issue by sharkeylaser** [chrismaltby/gb-studio#1520]
**Describe the bug**
When running GB Studio in the terminal, I get the following error when cancelling the build:
```
Error occurred in handler for 'project:build': Error: BUILD_CANCELLED
    at progress (/tmp/.mount_GB StuMUFwpI/usr/lib/gb-studio/resources/app.asar/.webpack/main/index.js:2:1256735)
    at t.default (/tmp/.mount_GB StuMUFwpI/usr/lib/gb-studio/resources/app.asar/.webpack/main/index.js:2:944750)
    at async t.default (/tmp/.mount_GB StuMUFwpI/usr/lib/gb-studio/resources/app.asar/.webpack/main/index.js:2:903581)
    at async /tmp/.mount_GB StuMUFwpI/usr/lib/gb-studio/resources/app.asar/.webpack/main/index.js:2:1256581
    at async node:electron/js2c/browser_init:2:97934
```
I also noticed it takes quite a while for GB Studio to actually cancel building, and even when it does appear to cancel (the "Cancel Build" button turns back into the "Run" button, the spinning compile button turns back to the play button), LCC/SDCC/etc still appear in the background.

**To Reproduce**
1. Run GB Studio in a terminal.
2. Compile a project.
3. Hit the "Cancel Build" button.
4. Look at the output of the terminal.

**Expected behavior**
It should force stop SDCC/LCC, only show the play and Run buttons when it is fully cancelled and all temp files are removed, and no crash in the terminal.

**Platform (please complete the following information):**
 - OS: Arch Linux
 - All versions of GB Studio 4 and even some of the later GB Studio 3 versions

**Comment by chrismaltby** [comment]
Hi @sharkeylaser previously when cancelling the build process would wait for the currently running SDCC/LCC processes to finish. I've pushed an update to develop that will now cancel the compiler processes (and any child processes that they've spawned) to be killed immediately. This will be included in v4.1

That warning will still appear on the console though, not a huge amount I can do about it, it's shown by Electron and is non configurable from what I can tell, but it's just a warning that the build process Promise has rejected, and is handled elsewhere in the render process, so you can safely ignore it.

**Comment by sharkeylaser** [comment]
@chrismaltby thank you for looking into this. I am glad the error can be ignored and that it will cancel the build immediately :)
